app/pulp/app/viewsets/base.py

```diff
@@ -87,11 +106,9 @@ def register_with(cls, router):
                        'correctly subclass the Master ViewSet, and do both have endpoint_name '
                        'set to different values?').format(cls.__name__)
                 warnings.warn(msg, RuntimeWarning)
```

Comment:
You actually got bit by this warning while working on this. Should it be a RuntimeError instead? I'm thinking probably it should.